### PR TITLE
RavenDB-12946 Restore CheckIfIdAlreadyIncluded for load with includes

### PR DESCRIPTION
**Pull request by efratshenhar**

### CODE REVIEW

File: src/Raven.Client/Documents/Session/Operations/LoadOperation.cs [hunk 1]
```diff
@@ -52,7 +52,7 @@ public LoadOperation ById(string id)
             if (_ids == null)
                 _ids = new[] { id };
 
-            if (_session.IsLoadedOrDeleted(id))
+            if (_session.IsLoadedOrDeleted(id) && (_includes == null))
```
Comment (ayende) [inline src/Raven.Client/Documents/Session/Operations/LoadOperation.cs]: This does only part of the work. In particular, id doesn't check that the included documents are already there. 
If they are, we don't need to hit the server at all

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Client/Documents/Session/Operations/LoadOperation.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Client/Documents/Session/Operations/LoadOperation.cs]: This is to error prone. Sooner or later someone will do `ById` first and then `Include`. Need to move that logic to `CreateRequest` method.

Comment (maximburyak) [reply src/Raven.Client/Documents/Session/Operations/LoadOperation.cs]: @ayende this actually already works, not treated here, see gist: https://gist.github.com/maximburyak/af6438d1847fa999960b99d2dcd1576d
@ppekrol I think that this a topic of a seperate issue.

Comment (ayende) [reply src/Raven.Client/Documents/Session/Operations/LoadOperation.cs]: where is the code that actually handles this?

Comment (ppekrol) [reply src/Raven.Client/Documents/Session/Operations/LoadOperation.cs]: It's in the `CreateRequest` method. I will sit with @efratshenhar on Sunday and we will figure it out.

File: src/Raven.Client/Documents/Session/Operations/LoadOperation.cs [hunk 2]
```diff
@@ -12,11 +13,10 @@ internal class LoadOperation
         private readonly InMemoryDocumentSessionOperations _session;
         private static readonly Logger Logger = LoggingSource.Instance.GetLogger<LoadOperation>("Client");
 
-        private string[] _ids;
+        private readonly List<string> _ids = new List<string>();
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Session/Operations/LoadOperation.cs]: better to use array only, then you will not have to do `.ToArray()` in `CreateRequest`

**Review by ppekrol — CHANGES_REQUESTED** [review]

**Review by ppekrol — APPROVED** [review]

**Pull request merged**